pkg/asset/installconfig/ibmcloud/client.go

```diff
@@ -0,0 +1,398 @@
+package ibmcloud
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/networking-go-sdk/zonesv1"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/IBM/platform-services-go-sdk/resourcemanagerv2"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockgen -source=./client.go -destination=./mock/ibmcloudclient_generated.go -package=mock
+
+// API represents the calls made to the API.
+type API interface {
+	GetCISInstance(ctx context.Context, crnstr string) (*resourcecontrollerv2.ResourceInstance, error)
+	GetCustomImageByName(ctx context.Context, imageName string, region string) (*vpcv1.Image, error)
+	GetCustomImages(ctx context.Context, region string) ([]vpcv1.Image, error)
+	GetDNSZones(ctx context.Context) ([]DNSZoneResponse, error)
+	GetEncryptionKey(ctx context.Context, keyCRN string) (*EncryptionKeyResponse, error)
+	GetResourceGroups(ctx context.Context) ([]resourcemanagerv2.ResourceGroup, error)
+	GetResourceGroup(ctx context.Context, nameOrID string) (*resourcemanagerv2.ResourceGroup, error)
+	GetSubnet(ctx context.Context, subnetID string) (*vpcv1.Subnet, error)
+	GetVSIProfiles(ctx context.Context) ([]vpcv1.InstanceProfile, error)
+	GetVPC(ctx context.Context, vpcID string) (*vpcv1.VPC, error)
+	GetVPCZonesForRegion(ctx context.Context, region string) ([]string, error)
+	GetZoneIDByName(ctx context.Context, crn string, name string) (string, error)
+}
+
+// Client makes calls to the IBM Cloud API.
+type Client struct {
+	managementAPI *resourcemanagerv2.ResourceManagerV2
+	controllerAPI *resourcecontrollerv2.ResourceControllerV2
+	vpcAPI        *vpcv1.VpcV1
+	authenticator *core.IamAuthenticator
+}
+
+// cisServiceID is the Cloud Internet Services' catalog service ID.
+const cisServiceID = "75874a60-cb12-11e7-948e-37ac098eb1b9"
+
+// VPCResourceNotFoundError represents an error for a VPC resoruce that is not found.
+type VPCResourceNotFoundError struct{}
```

Comment:
@staebler This is my best shot at implementing an error type like you suggested. I'm still pretty fresh with Go, so please let me know if this isn't quite right.